geode-core/src/main/java/org/apache/geode/internal/cache/PartitionedRegion.java

```diff
@@ -463,6 +463,19 @@ public void setShutDownAllStatus(int newStatus) {
 
   private final Object indexLock = new Object();
 
+  private Set<ColocationListener> colocationListeners = new ConcurrentHashSet<>();
+
+  public void addColocationListener(ColocationListener colocationListener) {
+    if (colocationListener != null) {
+      colocationListeners.add(colocationListener);
+    }
+  }
+
+  public void removeColocationListener(ColocationListener colocationListener) {
```

Comment:
Why is remove never called?

Reply:
It was put here to keep methods in pairs [add - remove pairs]. We do not need to use the remove for our lucene modules because we always need them in our index file regions for computation of index repository.
We put in a comment in the code to explain this nature.

Thank you Darrel for the reviews